src/plugins/discover/public/application/main/components/no_results/no_results_suggestions/no_results_suggestion_when_time_range.tsx

```diff
@@ -12,23 +12,70 @@ import {
   EuiDescriptionList,
   EuiDescriptionListTitle,
   EuiDescriptionListDescription,
+  EuiLink,
 } from '@elastic/eui';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { AggregateQuery, Filter, Query } from '@kbn/es-query';
+import { useDiscoverServices } from '../../../../../hooks/use_discover_services';
+import { useFetchOccurrencesRange } from './use_fetch_occurances_range';
+
+export interface Props {
+  dataView: DataView;
+  query?: Query | AggregateQuery;
+  filters?: Filter[];
+}
+export const NoResultsSuggestionWhenTimeRange: React.FC<Props> = ({ dataView, query, filters }) => {
+  const services = useDiscoverServices();
+  const { data, uiSettings, timefilter } = services;
+
+  const { range: occurrencesRange, refetch } = useFetchOccurrencesRange({
+    dataView,
+    query,
+    filters,
+    services: {
+      data,
+      uiSettings,
+    },
+  });
+
+  const expandTimeRange = async () => {
+    const range = await refetch();
+    if (range?.from && range?.to) {
+      timefilter.setTime({ from: range.from, to: range.to });
+    }
+  };
+
+  if (!occurrencesRange?.from || !occurrencesRange?.to) {
```

Comment:
Love this improvement, just adding my 2 cents, maybe a loading state indicator  when the `from`-`to` range is retrieved would be nice for longer running queries or  slow networks

![Kapture 2022-12-14 at 10 55 38](https://user-images.githubusercontent.com/463851/207568392-32d65fac-84b2-404a-b99d-5cd6451982c3.gif)